contracts/test/bridge/tokenBridge.ts

```diff
@@ -0,0 +1,184 @@
+        ethers.utils.hexZeroPad(mockStarknetMessaging.address, 32),
+      ])
+
+      await proxy.connect(deployer).addImplementation(starkNetERC20Bridge.address, new_data, false)
+      await proxy.connect(deployer).upgradeTo(starkNetERC20Bridge.address, new_data, false)
+    })
+
+    it('Should add implementation and upgrade successfully', async () => {
+      await owner.invoke(tokenBridgeContract, 'set_l2_token', {
+        l2_token_address: linkTokenContract.address,
+      })
+      const { res: l2_address } = await tokenBridgeContract.call('get_l2_token', {})
+      expectAddressEquality(l2_address.toString(), linkTokenContract.address)
+    })
+
+    it('Should wrap contract and set L2 TokenBridge successfully', async () => {
+      const contract = await loadStarkgateContract('StarknetERC20Bridge')
+      newStarkNetERC20Bridge = await ethers.getContractAt(contract.abi, proxy.address)
+
+      const tx = await newStarkNetERC20Bridge.setL2TokenBridge(BigInt(tokenBridgeContract.address))
+      await expect(tx)
+        .to.emit(newStarkNetERC20Bridge, 'LogSetL2TokenBridge')
+        .withArgs(BigInt(tokenBridgeContract.address))
+    })
+
+    it('Test Set and Get function for L1 bridge address', async () => {
+      await owner.invoke(tokenBridgeContract, 'set_l1_bridge', {
+        l1_bridge_address: newStarkNetERC20Bridge.address,
+      })
+      const { res: l1_address } = await tokenBridgeContract.call('get_l1_bridge', {})
+      expectAddressEquality(l1_address.toString(), newStarkNetERC20Bridge.address)
+    })
+
+    it('Should setBalance to the token bridge', async () => {
+      await testERC20.setBalance(newStarkNetERC20Bridge.address, 1000)
+      const balance = await testERC20.balanceOf(newStarkNetERC20Bridge.address)
+      expect(balance).to.equal(1000)
+
+      await testERC20.setBalance(deployer.address, 10)
+      const balance2 = await testERC20.balanceOf(deployer.address)
+      expect(balance2).to.equal(10)
+    })
+
+    it('Should set Max total balance', async () => {
+      await newStarkNetERC20Bridge.setMaxTotalBalance(100000)
+      const totalbalance = await newStarkNetERC20Bridge.maxTotalBalance()
+      expect(totalbalance).to.equal(100000)
+    })
+
+    it('Should set Max deposit', async () => {
+      await newStarkNetERC20Bridge.setMaxDeposit(100)
+      const deposit = await newStarkNetERC20Bridge.maxDeposit()
+      expect(deposit).to.equal(100)
+    })
+
+    it('Should deposit to the L2 contract, L1 balance should be decreased by 2', async () => {
+      await testERC20.approve(newStarkNetERC20Bridge.address, 2)
+      await newStarkNetERC20Bridge.deposit(2, owner.starknetContract.address)
+
+      const balance = await testERC20.balanceOf(deployer.address)
+      expect(balance).to.equal(8)
+    })
+  })
+
+  describe('Test bridge from L2 to L1', function () {
+    it('Should flush the L1 messages so that they can be consumed by the L2.', async () => {
```

Comment:
Ok, so here is the test in another scope :D

Let's please move it to the L1 -> L2 scope/suite, and then here test the L2 -> L1 in a new scope/suite set of tests.